### DISCUSSION TITLE
Plotting Pole Figures with the method of Harmonic Series Expansion

### DISCUSSION THREAD

**Discussion by aniketchakrabarty** [mtex-toolbox/mtex#691] [Ask Anything]
Is there a way in MTEX to plot Pole Figures with the Method of Harmonic Series Expansion. The Pole Figure should appear like the one posted below:
![image](https://user-images.githubusercontent.com/73643457/104194415-52977180-5447-11eb-9487-9e896b029a9a.png)
In MTEX , while plotting the pole figures, it appears as that of below:
![001-PF](https://user-images.githubusercontent.com/73643457/104194349-372c6680-5447-11eb-844e-9df013b7185b.JPG)

**Comment by lfgmorales** [comment]
Hi



I am not entirely sure, but it seems that in your MTEX pole figure you are missing the sample symmetry that you seem to have in the pole figure calculated using OIM. Would that be the case?


have a look here


https://mtex-toolbox.github.io/DetectionOfSampleSymmetry.html


cheers



Luiz





Luiz F. G. Morales
ScopeM & Structural Geology and Tectonics Group
ETH Zürich
________________________________

**Comment by ralfHielscher** [comment]
Could you please explain what input data you have?

Ralf.

**Comment by aniketchakrabarty** [comment]
Dear sir,

I understand that my concept of crystallographic symmetry is somewhat not
very strong. I think I had missed out on the sample symmetry.

Type of Input data I have : ebsd data.
My material is FCC Aluminium. I am trying to plot the Pole Figures directly
from the data.

On Mon, Jan 11, 2021 at 8:36 PM Ralf Hielscher <notifications@github.com>
wrote:

> Could you please explain what input data you have?
>
> Ralf.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mtex-toolbox/mtex/discussions/691#discussioncomment-274592>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARR3LQJWNNXYILMSY6ZG5UDSZMHXHANCNFSM4V5UPKPA>
> .
>


-- 
*Regards,*

*Aniket Chakrabarty*
Ph.D Research Scholar,
Department of Aerospace Engineering and Applied Mechanics,
Indian Institute of Engineering Science and Technology, Shibpur
Howrah-711103
West Bengal
India

**Reply by ralfHielscher** [reply]
Actually, you do completely fine. You could make your pole figures more smooth using the option `'halfwidth',15*degree`.
What don't you like about the pole figures you obtained? What do you mean with 
> The Pole Figure should appear like the one posted below:

Ralf

**Comment by aniketchakrabarty** [comment]
Dear sir,

I think the Pole Figures appearing from MTEX look more like they are
derived from ODFs (speaking very non-technically. Please forgive me for my
ignorance in this case). They do not have specified regions of contours.
When compared to another ebsd data analysis software (TSL-OIM), they use a
method called Harmonic Series Expansion for plotting pole figures, which
provides us with perfectly defined regions of contours.

The problem gets solved by specifying sample symmetry. For instance
ss=specimenSymmetry('222'). (As I am working with Aluminium, I assume that
my sample symmetry is orthorhombic and I have specified, '222').

**Comment by ralfHielscher** [comment]
> I think the Pole Figures appearing from MTEX look more like they are
derived from ODFs (speaking very non-technically. Please forgive me for my
ignorance in this case). They do not have specified regions of contours.
When compared to another ebsd data analysis software (TSL-OIM), they use a
method called Harmonic Series Expansion for plotting pole figures, which
provides us with perfectly defined regions of contours.

There are three different concepts which have nothing to do with each other
1. how the ODF is computed, e.g. from EBSD data, from XRD data, simulated, ...
2. how the ODF is stored, e.g. as harmonic series expansion, as discrete points, ...
3. how the ODF is plotted, by contour lines, as smooth plot, ....

In all three points, MTEX supports all the examples given above and many more. From your question 
I understand that you do not like the MTEX plot because it is not as smooth as you suspected it to be.
You can smooth your ODF by doing
```matlab
odf = smooth(odf)
```

You can also get a smoother plot by reducing the number of contour lines, e.g. 
```matlab
plotPDF(odf,Miller(1,0,0,odf.CS),'contourf','contours',5)
```
You should be careful smoothing your ODF, at the end it should represent reality right?

> The problem gets solved by specifying sample symmetry. For instance
ss=specimenSymmetry('222'). (As I am working with Aluminium, I assume that
my sample symmetry is orthorhombic and I have specified, '222').

The question is whether your your sample has indeed the imposed specimen symmetry. If you have measured a sufficiently portion of your specimen the specimen symmetry should become visible also without imposing it.

Ralf.
